src/com/t_oster/visicut/gui/MainView.java

```diff
@@ -1789,7 +1798,96 @@ private void editMappings() throws FileNotFoundException, IOException
     }
   }
 
-
+	private void generateGcodeMenuItemActionPerformed( java.awt.event.ActionEvent evg ) {
+		String jobname = getJobName();
+		List<String> warnings = new LinkedList<String>();
+		final Map<LaserProfile, List<LaserProperty>> cuttingSettings = this.getPropertyMapForCurrentJob();
+		if (cuttingSettings == null)
+		{
+			return;
+		}
+
```

Comment:
Here we should ask the user, where to save the file. One thing, we did not consider yet is the file extension (not every driver outputs gcode). To make it perfect, the driver should give the file extension (e.g. via a method getOutputFileExtension())... but right now I could live with ".dat" or something like this.